File: A. Technical Support/solve.cpp
#include <iostream>
using namespace std;

int main(){
	
	int n,a;
	cin >> n;
	while(n--){
		cin >> a;
		string s;
		cin >> s;

		int q=0;
		int a=0;

		for (char c: s){
			if (c == 'Q'){
				q++;
			}
			else{
				if (q > 0){
					q--;
				}
			}
		}

		if (q){
			cout << "No" << endl;
		}
		else{
			cout << "Yes" << endl;
		}


	}
	
		

	return 0;
}

File: Neat Brackets Toph/solve.py
a = input()
ct = 0
b1,b2 = 0,0

for i in a:
	if i == "(":
		ct+=1
		b1+=1
	else:
		if ct > 0:
			ct -= 1
		b2+=1

if b1 != b2:
	ct+=1

print("Yes" if ct == 0 else "No")
